developer_guides/testbench_fuzzing/testbench_afl_fuzzing.rst

```diff
@@ -0,0 +1,96 @@
+Fuzzing testbench using AFL fuzzer
+==================================
+
+Install AFL fuzzer
+------------------
+
+Follow the steps from the `Quick Start
+Guide <https://github.com/google/AFL/blob/master/docs/QuickStartGuide.txt>`__
+from AFL repository to install the AFL fuzzer in your system.
+
+From this point onwards, i assume that installation directory of the AFL
+fuzzer is at
+
+::
+
+   $HOME/work/
+
+Building testbench with AFL instrumentation
+-------------------------------------------
+
+AFL is a brute-force fuzzer with an exceedingly simple but rock-solid
+instrumentation guided algorithm. So, it's important that we add
+instrumentation to the code before we run fuzzer on it to get good
+results.
+
+When you build AFL from the previous step, an **afl-gcc** executable is
+generated, this works as a companion tool that acts as a drop-in
+replacement for gcc or clang.
+
+So, before we build testbench we need to make sure we are compiling our
+code with **afl-gcc** in order to add instrumentation to the code.
+
+**The host-build-all.sh script from the scripts/ directory does exactly
+this when** **you run it with -f option.**
+
+**Note**: If you have installed the AFL in any other directory, you need
+to change the path in the host-build-all.sh script appropriately. By
+default, the script assumes you have installed AFL in '*home/sof/work*'
```

Comment:
do we assume ```$HOME/sof``` or ```home/sof``` since they are different.

Reply:
It's '$HOME' only. I missed this. Thanks!